chx_compress/io/multifile/multifile.py

```diff
@@ -249,13 +249,44 @@ def multifile_reader(filepath, mode="rb"):
     return mfr
 
 
-def get_dense_array(multifile_reader):
+def get_dense_image(multifile_reader, image_index, target_image_array):
+    """A convenience function to build a dense image array.
+
+    The target image array should have shape (rows, columns) or
+    (columns, rows) depending on the file being read. The latter
+    is standard practice at CHX.
+
+    Parameters
+    ----------
+    multifile_reader : MultifileReader
+
+    image_index : int
+      index of the image to return
+
+    target_image_array : ndarray
+      (rows, columns) or (columns, rows) array for image data, will be overwritten
+
+    Return
+    ------
+    dense_array : ndarray
+      dense array representation of the multifile data
+
+     """
+    target_image_array[:] = 0
```

Comment:
Why do you want to pass in an array to overwrite?

Reply:
For performance and so the caller can decide COLUMNSxROWS or ROWSxCOLUMNS.

Reply:
By "for performance" I mean the caller can decide to reuse one image array or create a large array to hold many images. Even if in most cases there is one "good" choice I don't want to make a caller work hard to get this level of control. Deciding what that image array should look like is not something I want this code to do.